screens/AppNavigator/screens/Dex/DexAddLiquidity.tsx

```diff
@@ -0,0 +1,225 @@
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpair'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { useCallback, useState } from 'react'
+import { TouchableOpacity } from 'react-native'
+import { ScrollView } from 'react-native-gesture-handler'
+import tailwind from 'tailwind-rn'
+import { Text, TextInput, View } from '../../../../components'
+import { getTokenIcon } from '../../../../components/icons/tokens/_index'
+import { PrimaryColorStyle } from '../../../../constants/Theme'
+import { useTokensAPI } from '../../../../hooks/wallet/TokensAPI'
+import { DexParamList } from './DexNavigator'
+
+type Props = StackScreenProps<DexParamList, 'AddLiquidity'>
+type EditingAmount = 'primary' | 'secondary'
+
+interface ExtPoolPairData extends PoolPairData {
+  aSymbol: string
+  bSymbol: string
+  aToBRate: BigNumber
+  bToARate: BigNumber
+}
+
+export function AddLiquidityScreen (props: Props): JSX.Element {
+  // TODO: poll for PoolPairData periodically
+
+  const navigation = useNavigation<NavigationProp<DexParamList>>()
+
+  // this component state
+  const [tokenAAmount, setTokenAAmount] = useState<BigNumber>(new BigNumber(0))
+  const [tokenBAmount, setTokenBAmount] = useState<BigNumber>(new BigNumber(0))
+  // ratio, before times 100
+  const [sharePercentage, setSharePercentage] = useState<BigNumber>(new BigNumber(0))
+
+  // gather required data
+  const tokens = useTokensAPI()
+  const { pair: poolPairData } = props.route.params
+  const [aSymbol, bSymbol] = poolPairData.symbol.split('-')
+  const pair: ExtPoolPairData = {
+    ...poolPairData,
+    aSymbol,
+    bSymbol,
+    aToBRate: new BigNumber(poolPairData.tokenB.reserve).div(poolPairData.tokenA.reserve),
+    bToARate: new BigNumber(poolPairData.tokenA.reserve).div(poolPairData.tokenB.reserve)
+  }
+  const addressTokenA = tokens.find(at => at.id === pair.tokenA.id)
+  const addressTokenB = tokens.find(at => at.id === pair.tokenB.id)
+  // expected behavior: address token is not listed when no token
+  const balanceA = addressTokenA === undefined ? new BigNumber(0) : new BigNumber(addressTokenA.amount)
```

Comment:
This might be problematic in the future if we add polling on this page. You should store this on a useState instead of just putting it as a variable

Reply:
Did a quick test on this. These variable are derived from `useTokensAPI()` result (L39), so it does update when fetchTokens triggered in the background. Indeed, adding another useState here, the setState will trigger unnecessary re-render.